src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesLauncher.java

```diff
@@ -166,12 +165,13 @@ else if (httpCode == 409 && e.getMessage().contains("Operation cannot be fulfill
             ObjectMeta podMetadata = pod.getMetadata();
             template.getWorkspaceVolume().createVolume(client, podMetadata);
             template.getVolumes().forEach(volume -> volume.createVolume(client, podMetadata));
-            watcher = new AllContainersRunningPodWatcher(client, pod);
-            try (Watch w1 = client.pods().inNamespace(namespace).withName(podName).watch(watcher);
-                 Watch w2 = eventWatch(client, podName, namespace, runListener)) {
-                assert watcher != null; // assigned 3 lines above
-                watcher.await(template.getSlaveConnectTimeout(), TimeUnit.SECONDS);
+
+            // Removed watcher because of load issues making the fabric8 library wait indefinitely https://issues.jenkins.io/browse/JENKINS-68126
+            Pod checkPod = client.pods().inNamespace(namespace).withName(podName).waitUntilReady(template.getSlaveConnectTimeout(), TimeUnit.SECONDS);
+            if (!Readiness.isPodReady(checkPod)) {
+                throw new KubernetesClientTimeoutException(pod, template.getSlaveConnectTimeout(), TimeUnit.SECONDS);
             }
```

Comment:
I'm not sure this is even necessary, the exception should be raised already by `waitUntilReady` if it reaches the timeout.

Reply:
I'm not sure...